### PR TITLE
Added watchNamespacedResource

### PR DESCRIPTION
**Pull request by wittfeldt**
Love the library! I needed to watch a namespaced custom resource so I added a method for it. Maybe there is a better way?

### CODE REVIEW

**Comment by pfisterer** [comment]
@wittfeldt tnx for the PR, I'm also currently evaluating this library 

- Operator: https://github.com/pfisterer/edsc-dns-domains/blob/master/src/dnssec-operator.js
- CRD yaml: https://github.com/pfisterer/edsc-dns-domains/blob/master/src/crd-defs/dnssec-zone-crd-v1beta1.yaml
- Example resource: https://github.com/pfisterer/edsc-dns-domains/blob/master/test/example-record.yaml

However, when `kubectl apply -f`ing the example resource, I do not get any callback. 

@wittfeldt @dot-i Do you have a working example that you can share?

**Comment by dot-i** [comment]
@pfisterer just to understand clearly: is your question based on the new code in this PR, or a general question? 😄

File: src/operator.ts [hunk 1]
```diff
@@ -211,6 +211,51 @@ export default abstract class Operator {
...
 
+    /**
+     * Watch a namespaced Kubernetes resource.
+     * @param group The group of the resource or an empty string for core resources
+     * @param version The version of the resource
+     * @param plural The plural name of the resource
+     * @param namespace The namespace which the resource is in
+     * @param onEvent The async callback for added, modified or deleted events on the resource
+     */
+    protected async watchNamespacedResource(group: string, version: string, plural: string, onEvent: (event: ResourceEvent) => Promise<void>, namespace?: string): Promise<void> {
```
Comment (dot-i) [inline src/operator.ts:222]: I would just modify the existing `watchResource()` to include the new optional `namespace?` parameter. It's backwards compatible after all.

Also make sure to update `README.md` with the new parameter.

File: .gitignore [hunk 2]
```diff
@@ -107,5 +107,3 @@ typings/
 # End of https://www.gitignore.io/api/node,visualstudiocode
 
 # Custom rules (everything added below won't be overriden by 'Generate .gitignore File' if you use 'Update' option)
-
-dist
```
Comment (dot-i) [inline .gitignore:111]: This needs to be excluded again, not sure why it caused you issues 🤔

**Comment by pfisterer** [comment]
@dot-i Tnx for your reply. A bit of both because I'd like to watch namespaced custom resources and would love to see an example of how everything fits together. But I'm still struggling with getting the non-PR code to work (see the links above).

File: src/operator.ts (hunk 1 above)
Comment (wittfeldt) [reply src/operator.ts:222]: You're right, I will update the MR

File: .gitignore (hunk 2 above)
Comment (wittfeldt) [reply .gitignore:111]: Oups, I forgot that I created this MR and pushed some last-minute fixes required for our internal build system.

**Comment by wittfeldt** [comment]
> @wittfeldt @dot-i Do you have a working example that you can share?

@pfisterer I created a gist that I hope can help you to get started: https://gist.github.com/wittfeldt/29a87a6ee1a3975c991dcfc2276e1ba3

**Comment by wittfeldt** [comment]
Closing due to https://github.com/dot-i/k8s-operator-node/pull/3

**Comment by pfisterer** [comment]
Great, thanks a lot for your support @wittfeldt!

**Comment by pfisterer** [comment]
Actually I had pretty much the same code except for code throwing an exception which was swallowed somewhere else. Anyway, comparing yours to mine was very helpful. Thank you again.

**Comment by wittfeldt** [comment]
@pfisterer You're welcome. Good luck with your project

**Pull request closed without merging**